libbeat/beat/beat.go

```diff
@@ -65,15 +67,36 @@ func NewBeat(name string, version string, bt Beater) *Beat {
 		Name:    name,
 		BT:      bt,
 		UUID:    uuid.NewV4(),
+
+		done: make(chan struct{}),
 	}
 
 	return &b
 }
 
 // Initiates and runs a new beat object
-func Run(name string, version string, bt Beater) *Beat {
+func Run(name string, version string, bt Beater) {
 	b := NewBeat(name, version, bt)
 
+	// Runs beat inside a go process
+	go func() {
+		logp.Info("Start Beat")
+		b.Start()
+		b.Stop()
```

Comment:
I'm also not too big of a fan to call it twice, but I would prefer that actually the stop function make sure it is only called once and if it is called a second time, it just skips. Or that Stop only happens, if beater is actually running.